controllers/imageupdateautomation_controller.go

```diff
@@ -78,22 +80,41 @@ type ImageUpdateAutomationReconciler struct {
 func (r *ImageUpdateAutomationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("imageupdateautomation", req.NamespacedName)
+	now := time.Now()
 
 	var auto imagev1.ImageUpdateAutomation
 	if err := r.Get(ctx, req.NamespacedName, &auto); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	auto.Status.LastAutomationRunTime = &metav1.Time{Time: now}
+	if err := r.Status().Update(ctx, &auto); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// failWithError is a helper for bailing on the reconciliation.
+	failWithError := func(err error) (ctrl.Result, error) {
```

Comment:
Maybe this should be extracted to a reconciler method, which would make for a shorter Reconcile func body.

Reply:
I considered that, but thought the convenience of closing over a few of the local variables was worth the extra assignment here.